cmd/podman/containers/cp.go

```diff
@@ -113,6 +116,82 @@ func doCopy(funcA func() error, funcB func() error) error {
 	return errorhandling.JoinErrors(copyErrors)
 }
 
+func copyContainerToContainer(fromContainer string, sourcePath string, toContainer string, destPath string) error {
+	if err := containerMustExist(fromContainer); err != nil {
+		return err
+	}
+
+	if err := containerMustExist(toContainer); err != nil {
+		return err
+	}
+
+	fromContainerInfo, err := registry.ContainerEngine().ContainerStat(registry.GetContext(), fromContainer, sourcePath)
+	if err != nil {
+		return errors.Wrapf(err, "%q could not be found on container %s", sourcePath, fromContainer)
+	}
+
+	var toContainerBaseName string
+	_ = toContainerBaseName
+	toContainerInfo, err := registry.ContainerEngine().ContainerStat(registry.GetContext(), toContainer, destPath)
+	if err != nil {
+		if strings.HasSuffix(destPath, "/") {
+			return errors.Wrapf(err, "%q could not be found on container %s", destPath, toContainer)
+		}
+		if toContainerInfo != nil {
+			toContainerBaseName = filepath.Base(toContainerInfo.LinkTarget)
+		} else {
+			toContainerBaseName = filepath.Base(destPath)
+		}
+
+		parentDir, err := containerParentDir(toContainer, destPath)
+		if err != nil {
+			return errors.Wrapf(err, "could not determine parent dir of %q on container %s", destPath, toContainer)
+		}
+		toContainerInfo, err = registry.ContainerEngine().ContainerStat(registry.GetContext(), toContainer, parentDir)
+		if err != nil {
+			return errors.Wrapf(err, "%q could not be found on container %s", destPath, toContainer)
+		}
+	} else {
+		toContainerBaseName = filepath.Base(toContainerInfo.LinkTarget)
+	}
+
+	if fromContainerInfo.IsDir && !toContainerInfo.IsDir {
+		return errors.New("destination must be a directory when copying a directory")
+	}
+
+	fromContainerTarget, toContainerTarget := fromContainerInfo.LinkTarget, toContainerInfo.LinkTarget
+	if !toContainerInfo.IsDir {
+		toContainerTarget = filepath.Dir(destPath)
+	}
+	reader, writer := io.Pipe()
+
+	fromContainerCopy := func() error {
```

Comment:
potentially could consolidate `toContainerCopy` and `fromContainer` copy into a function you call twice that takes specific parameters. Seems slightly redundant to have nearly identical code twice.

Reply:
there might be some changes to the functions, so depending on what the final outcome looks like i'll refactor the code